### ISSUE TITLE
[1.3-rc.12] Sometimes modules don't get export from old-style packages

### ISSUE THREAD

**Issue by clayne11** [meteor/meteor#6590]
When I use an old-style Meteor package (aka no `api.mainModule`), for some packages `import` doesn't see any exports.

In these instances, when I try to `import` something that is available globally from that package, it returns undefined.

Example:

```
// package 1
Package.describe({
  name: 'users',
  version: '0.0.1',
});

Package.onUse(function(api) {
  api.use('ecmascript');
  api.addFiles('users.js');
  api.export('Users');
});

// users.js
Users = 'userString';

// package 2
Package.describe({
  name: 'test',
  version: '0.0.1',
});

Package.onUse(function(api) {
  api.use('ecmascript');
  api.use('users');
  api.addFiles('test.js');
});

// test.js (attempt 1)
import {Users} from 'meteor/users';

console.log(Users) // undefined

// test.js (attempt 2 - sanity check)
console.log(Users) // userString
```

This only seems to happen some of the time and I'm not sure what the difference is when this happens vs when it works as expected.

**Comment by benjamn** [comment]
Can you try putting together a reproduction repo, even if it doesn't reproduce the problem totally reliably?

**Comment by Zaiban** [comment]
I think the error I'm having is maybe related to this. Trying to use npm installed Mocha for unit testing, but running the tests gives errors like:
`Error: Cannot find module 'meteor/meteor'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (methods.js:1:1)
    at Module._compile (module.js:413:34)
    at loader (root/node_modules/babel-register/lib/node.js:126:5)
    at Object.require.extensions.(anonymous function) [as .js] (root/node_modules/babel-register/lib/node.js:136:7)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (methods.js:1:1)
    at Module._compile (module.js:413:34)
    at loader (root/node_modules/babel-register/lib/node.js:126:5)
    at Object.require.extensions.(anonymous function) [as .js] (root/node_modules/babel-register/lib/node.js:136:7)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (account.js:1:1)
    at Module._compile (module.js:413:34)
    at loader (root/node_modules/babel-register/lib/node.js:126:5)
    at Object.require.extensions.(anonymous function) [as .js] (root/node_modules/babel-register/lib/node.js:136:7)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (account.tests.js:3:1)
    at Module._compile (module.js:413:34)
    at loader (root/node_modules/babel-register/lib/node.js:126:5)
    at Object.require.extensions.(anonymous function) [as .js] (root/node_modules/babel-register/lib/node.js:136:7)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at root/node_modules/mocha/lib/mocha.js:219:27
    at Array.forEach (native)
    at Mocha.loadFiles (root/node_modules/mocha/lib/mocha.js:216:14)
    at Mocha.run (root/node_modules/mocha/lib/mocha.js:468:10)
    at loadAndRun (root/node_modules/mocha/bin/_mocha:359:22)
    at Object.<anonymous> (root/node_modules/mocha/bin/_mocha:376:3)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Function.Module.runMain (module.js:447:10)
    at startup (node.js:141:18)
    at node.js:933:3
`
..when it is trying to import file that contains
`import { Meteor } from 'meteor/meteor';`

Same thing happens with all 'meteor/xxx' imports.

**Comment by benjamn** [comment]
@Zaiban in your case it looks like Mocha is trying to run a file in pure Node, outside the Meteor module system, which is probably why `meteor/...` packages aren't importable. Are you running `mocha` from the command line, or as part of a Meteor app?

**Comment by clayne11** [comment]
@benjamn to be clear it happens with certain packages 100% of the time. It's not an intermittent issue. I'll try to create a repro but I'm really not sure what causes certain packages to behave in this way.

**Comment by clayne11** [comment]
I was able to reproduce it.

https://github.com/clayne11/meteor-legacy-package-issue

If you run that project you'll see that `TestOne` prints undefined, but if you remove the `import` statement it print `TestOne`.

**Comment by tmeasday** [comment]
This does seem strange. I can't see any obvious reason for it (why is `test:package-one` compiled differently to the `meteor` package?)

**Comment by clayne11** [comment]
My app has about 115 packages that I've created and I've only encountered this issue on a handful of them so far.

**Comment by tmeasday** [comment]
Well if you can figure out what's different that'll be super helpful @clayne11

**Comment by benjamn** [comment]
This issue will be officially fixed when Meteor 1.3.2 is released today or tomorrow, but you can test it out by doing `meteor update --release 1.3.2-rc.5` now.
